File: zenoh-java/src/commonMain/kotlin/io/zenoh/query/ReplyKeyExpr.kt
//
// Copyright (c) 2026 ZettaScale Technology
//
// This program and the accompanying materials are made available under the
// terms of the Eclipse Public License 2.0 which is available at
// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
// which is available at https://www.apache.org/licenses/LICENSE-2.0.
//
// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
//
// Contributors:
//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
//

package io.zenoh.query

/**
 * The key expressions accepted by a query reply.
 *
 * Controls whether a GET query accepts replies whose key expressions
 * don't match the query's key expression.
 */
enum class ReplyKeyExpr {

    /**
     * Accept replies whose key expressions match the query key expression.
     */
    MATCHING_QUERY,

    /**
     * Accept replies whose key expressions may not match the query key expression.
     */
    ANY;
}

File: zenoh-flat/src/query/reply_key_expr.rs
use prebindgen_proc_macro::prebindgen;

/// Whether a query accepts replies whose key expression may not match the
/// query's. Mirrors `zenoh::query::ReplyKeyExpr`.
#[prebindgen]
#[repr(i32)]
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum ReplyKeyExpr {
    Any = 0,
    MatchingQuery = 1,
}

impl From<zenoh::query::ReplyKeyExpr> for ReplyKeyExpr {
    fn from(r: zenoh::query::ReplyKeyExpr) -> Self {
        match r {
            zenoh::query::ReplyKeyExpr::Any => ReplyKeyExpr::Any,
            zenoh::query::ReplyKeyExpr::MatchingQuery => ReplyKeyExpr::MatchingQuery,
        }
    }
}

impl From<ReplyKeyExpr> for zenoh::query::ReplyKeyExpr {
    fn from(r: ReplyKeyExpr) -> Self {
        match r {
            ReplyKeyExpr::Any => zenoh::query::ReplyKeyExpr::Any,
            ReplyKeyExpr::MatchingQuery => zenoh::query::ReplyKeyExpr::MatchingQuery,
        }
    }
}
